plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java

```diff
@@ -466,6 +468,9 @@ else if (varcharType.getBoundedLength() <= MEMSQL_MEDIUMTEXT_MAX_LENGTH) {
         if (REAL.equals(type)) {
             return WriteMapping.longMapping("float", realWriteFunction());
         }
+        if (DATE.equals(type)) {
+            return WriteMapping.longMapping("date", dateWriteFunctionUsingLocalDate());
+        }
```

Comment:
btw can we remove `legacyToWriteMapping` below?

Reply:
I guess so. Let me handle in follow-up PR.